### PR TITLE
more cli interactivity, bugfix

### PR DESCRIPTION
**Pull request by efriis**

### CODE REVIEW

**Comment by efriis** [comment]
fyi @eyurtsev

File: libs/cli/langchain_cli/namespaces/app.py [hunk 1]
```diff
@@ -55,21 +54,55 @@ def new(
...
+            raise typer.BadParameter("name is required when --nointeractive is set")
+        name_str = name
+        pip_bool = bool(pip)
+    else:
+        name_str = (
+            name if name else typer.prompt("What folder would you like to create?")
         )
+        if not has_packages:
+            package = []
+            package_prompt = "What package would you like to add? (leave blank to skip)"
```
Comment (eyurtsev) [inline libs/cli/langchain_cli/namespaces/app.py:82]: We don't have a search interface here right? Would it be possible to add example syntax for packages? (I actually don't know the answer to this, but are the only supported packages the one's in templates directory or is there a notation for packages that are located elsewhere?)

File: libs/cli/langchain_cli/namespaces/app.py [hunk 2]
```diff
@@ -55,21 +54,55 @@ def new(
...
+                package.append(package_str)
+                package_prompt = (
+                    f"{len(package)} added. Any more packages (leave blank to end)?"
+                )
+
+            has_packages = len(package) > 0
+
+        pip_bool = False
+        if pip is None and has_packages:
+            pip_bool = typer.confirm(
```
Comment (eyurtsev) [inline libs/cli/langchain_cli/namespaces/app.py:98]: I bet that a lot of users will not know what this means. Could we include an explanation about which default we recommend?

**Review by eyurtsev — APPROVED** [review]

File: libs/cli/langchain_cli/namespaces/app.py [hunk 3]
```diff
@@ -55,21 +54,55 @@ def new(
             is_flag=True,
         ),
     ] = None,
+    nointeractive: Annotated[
+        bool,
+        typer.Option(
+            "--nointeractive",
```
[the remark this answers is not in the document]
Comment (efriis) [reply libs/cli/langchain_cli/namespaces/app.py:60]: I like `--non-interactive` or `--quiet` maybe?

File: libs/cli/langchain_cli/namespaces/app.py (hunk 1 above)
Comment (efriis) [reply libs/cli/langchain_cli/namespaces/app.py:82]: Not yet. And can add packages in either of these formats (from `langchain app add --help` which I realize I need to fix newlines in)
```
langchain app add extraction-openai-functions 
langchain app add \
 git+ssh://git@github.com/efriis/simple-pirate.git
```

File: libs/cli/langchain_cli/namespaces/app.py (hunk 2 above)
Comment (efriis) [reply libs/cli/langchain_cli/namespaces/app.py:98]: Could also construct the exact command before this, and ask
```
To install these {n} templates with `pip`, run:

pip install -e ...

Would you like to run it? [y/N]
```
Which defaults to no `[y/N]`

File: libs/cli/langchain_cli/namespaces/app.py (hunk 1 above)
Comment (efriis) [reply libs/cli/langchain_cli/namespaces/app.py:82]: Added `langchain template list` which you can also search with e.g. `langchain template list rag`

File: libs/cli/langchain_cli/namespaces/app.py (hunk 2 above)
Comment (efriis) [reply libs/cli/langchain_cli/namespaces/app.py:98]: This is a bigger refactor. Will just rephrase as "Would you like to install these templates in your environment with pip?"

**Pull request merged**